impl/src/expand.rs

```diff
@@ -37,7 +37,7 @@ fn fallback(input: &DeriveInput, error: syn::Error) -> TokenStream {
 
         #[allow(unused_qualifications)]
         #[automatically_derived]
-        impl #impl_generics std::error::Error for #ty #ty_generics #where_clause
+        impl #impl_generics ::thiserror::__private::error::Error for #ty #ty_generics #where_clause
```

Comment:
Out of interest, why did we choose this rather than hardcoding `::std` or `::core`, in the compiled output with e.g. an `#error_namespace` variable?

My slight concern is that use of `::thiserror::__private` may leak this `__private` namespace out and could confuse people debugging this. But using `::std` / `::core` directly it's more obvious what's failing.